framework/mutex/OracleMutex.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\mutex;
+
+use PDO;
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * OracleMutex implements mutex "lock" mechanism via Oracle locks.
+ *
+ * Application configuration example:
+ *
+ * ```
+ * [
+ *     'components' => [
+ *         'db' => [
+ *             'class' => 'yii\db\Connection',
+ *             'dsn' => 'oci:dbname=LOCAL_XE',
+ *              ...
+ *         ]
+ *         'mutex' => [
+ *             'class' => 'yii\mutex\OracleMutex',
+ *              ...
+ *         ],
+ *     ],
+ * ]
+ * ```
+ *
+ * @see Mutex
+ *
+ * @author Alexander Zlakomanov <zlakomanoff@gmail.com>
+ * @since 1.0
+ */
+class OracleMutex extends DbMutex
+{
+    /**
+     * @var string driver name.
+     */
+    public $driverName = 'oci';
+
+    /**
+     * @var array available lock modes
+     */
+    private $availableLockModes = [
+        'X_MODE',
+        'NL_MODE',
+        'S_MODE',
+        'SX_MODE',
+        'SS_MODE',
+        'SSX_MODE'
+    ];
+
+    /**
+     * Initializes Oracle specific mutex component implementation.
+     * @throws InvalidConfigException if [[db]] is not MySQL connection.
+     */
+    public function init()
+    {
+        parent::init();
+        if ($this->db->driverName !== $this->driverName) {
+            throw new InvalidConfigException('In order to use OracleMutex connection must be configured to use Oracle database.');
+        }
+    }
+
+    /**
+     * Acquires lock by given name.
+     * @param string $name of the lock to be acquired.
+     * @param integer $timeout to wait for lock to become released.
+     * @param string $lockMode lock mode.
+     * @param boolean $releaseOnCommit release on commit.
+     * @return boolean acquiring result.
+     * @see http://docs.oracle.com/cd/B19306_01/appdev.102/b14258/d_lock.htm
+     */
+    protected function acquireLock($name, $timeout = 0, $lockMode = 'X_MODE', $releaseOnCommit = false)
```

Comment:
It seems you do not listen.
Method `acquireLock()` is **protected**, thus it can NOT be called from outside the mutex component:

```php
Yii::$app->mutex->acquireLock('mylock'); // PHP fatal error!
```